docs/source/impl-guide/normalization.rst

```diff
@@ -3,50 +3,105 @@
 Normalization
 !!!!!!!!!!!!!
 
-Certain insertion or deletion alleles may be represented ambiguously
-when using conventional sequence normalization, resulting in
-significant challenges when comparing such alleles.
+In VRS, "normalization" refers to the process of converting a given
+variation representation into a canonincal form.  Left- and
+right-shuffling are examples of normalization for sequence variants.
+VRS extends this concept to apply to all classes of VRS Variation.
+VRS normalization minimizes a class of ambiguity that impedes
+comparison of variation across systems.
 
-VRS describes a "fully-justified" normalization algorithm
-inspired by NCBI's Variant Overprecision Correction Algorithm [1]_.
+Implementations MUST provide a normalize function that accepts any
+Variation object and returns a normalized Variation.  Guidelines for
+these functions are below.
+
+
+General Normalization Rules
+@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+* VRS normalization functions are homomorphic: That is, the input and
+  output objects are always of the same `type`.
+* Object types that do not have VRS normalization rules are returned
+  as-is.  That is, all types of Variation MUST be supported, even if
+  such objects are unchanged.
+* VRS normalization functions are idempotent: Normalizing a normalized
+  Variation object returns an object with the same data.
+
+
+Allele Normalization
+@@@@@@@@@@@@@@@@@@@@
+
+Certain insertion or deletion alleles may have ambiguous
+representations when using conventional sequence normalization,
+resulting in significant challenges when comparing such alleles.
+
+VRS uses a "fully-justified" normalization algorithm inspired by
+NCBI's Variant Overprecision Correction Algorithm [1]_.
 Fully-justified normalization expands such ambiguous representation
 over the entire region of ambiguity, resulting in an *unambiguous*
 representation that may be readily compared with other alleles.
 
-VRS RECOMMENDS that Alleles at precise locations are
-normalized to a fully justified form unless there is a compelling
-reason to do otherwise.
+VRS RECOMMENDS that Alleles at precise locations are normalized to a
+fully justified form unless there is a compelling reason to do
+otherwise.  In addition, Alleles MUST be normalized in order to
```

Comment:
@reece 

> ..Alleles MUST be normalized in order to generate computed identifiers.

Are we saying that it's not possible to generate computed identifiers from non-normalized alleles or are we saying that generated computed identifiers from non-normalized alleles are not spec compliant?
"

Reply:
My inclination was to permit identification of non-normalized alleles. Alex wants computed identifiers limited to normalized alleles. I agreed that the it's easier to loosen than restrict over time, we three (I think?) agreed to require normalization now and and consider relaxing that later. (Even writing this, I have some serious reservations about this direction.)

Reply:
Yes. We all agreed to REQUIRE normalization (on all VRS objects, not just Alleles) to prevent people from treating normalization as an optional process, and specifically in the case of Alleles this is to prevent implementations from representing insertions / deletions that are left or right shifted (as seen in other variant representation standards).

In my view, VRS should either specify a normalization policy and adhere to it, or should only specify the message structure and leave normalization out-of-scope. Every rule and recommendation added to the specification adds complexity, and that cost should come with the benefit of enforcing / improving interoperability. If there are serious reservations about requiring normalization before generating computed IDs, then we should also be asking ourselves if normalization even belongs in the spec, or if it should be part of a separate policy document.

Reply:
In a subsequent meeting, we (Larry, Alex, Reece) agreed that normalization needed to have an option to preserve flanking sequence around a variant in order to be consistent with the representation of an allele of reference agreement sequence, but that this optionality undermines a desire to require normalization.

Therefore, we agreed that variation SHOULD be normalized before digesting, and that we would add an explanation in Design Decisions. I'll work on that addition to this PR.

Reply:
Change made in def5cd0.